### PR TITLE
[C-Api/Service] update API description

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Under the API review, update API description and example code.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

File: c/include/ml-api-service.h [hunk 1]
```diff
@@ -38,27 +38,63 @@ extern "C" {
...
+ * // Users may register intelligence pipelines for other processes and fetch such registered pipelines.
+ * // For example, a developer adds a pipeline which includes preprocessing and invoking a neural network model,
+ * // then an application can fetch and construct this for intelligence service.
+ * status = ml_service_get_pipeline ("my_pipeline", &pipeline);
+ * if (status != ML_ERROR_NONE) {
+ *   // handle error case
+ *   goto error;
+ * }
+ *
+ * status = ml_pipeline_construct (pipeline, NULL, NULL, &handle);
```
Comment (gichan-jang) [inline c/include/ml-api-service.h:74]:
```suggestion
 * status = ml_pipeline_construct (description, NULL, NULL, &handle);
```

File: c/include/ml-api-service.h [hunk 2]
```diff
@@ -38,27 +38,63 @@ extern "C" {
...
+ * if (status != ML_ERROR_NONE) {
+ *   // handle error case
+ *   goto error;
+ * }
+ *
+ * error:
+ * ml_pipeline_destroy (handle);
+ * g_free (pipeline);
+ * @endcode
```
Comment (gichan-jang) [inline c/include/ml-api-service.h:83]: It's optional but how about adding `ml_service_delete_pipeline` for the example.

File: c/include/ml-api-service.h [hunk 3]
```diff
@@ -38,27 +38,63 @@ extern "C" {
...
+ *
+ * status = ml_pipeline_construct (pipeline, NULL, NULL, &handle);
+ * if (status != ML_ERROR_NONE) {
+ *   // handle error case
+ *   goto error;
+ * }
+ *
+ * error:
+ * ml_pipeline_destroy (handle);
+ * g_free (pipeline);
```
Comment (gichan-jang) [inline c/include/ml-api-service.h:82]:
```suggestion
 * g_free (description);
```

File: c/include/ml-api-service.h [hunk 4]
```diff
@@ -38,27 +38,63 @@ extern "C" {
...
+ * if (status != ML_ERROR_NONE) {
+ *   // handle error case
+ *   goto error;
+ * }
+ *
+ * // Example to construct a pipeline with stored pipeline description.
+ * // Users may register intelligence pipelines for other processes and fetch such registered pipelines.
+ * // For example, a developer adds a pipeline which includes preprocessing and invoking a neural network model,
+ * // then an application can fetch and construct this for intelligence service.
+ * status = ml_service_get_pipeline ("my_pipeline", &pipeline);
```
[the remark this answers is not in the document]
Comment (jaeyun-jung) [reply c/include/ml-api-service.h:68]: better changing param name (gchar *description > pipeline)

**Review by again4you — APPROVED** [review]
LGTM!

**Review by myungjoo — APPROVED** [review]
LGTM

**Pull request merged**